tests/src/Kernel/ValueObject/ImageValueObjectTest.php

```diff
@@ -56,44 +63,63 @@ protected function setUp(): void {
 
     // Copy file in public files to use it for styling.
     \Drupal::service('file_system')->copy(__DIR__ . '/../../../fixtures/example_1.jpeg', 'public://example_1.jpg');
-  }
 
-  /**
-   * Test the image value object has the correct style applied.
-   */
-  public function testFromStyledImageItem() {
     // Create image file.
     $image = File::create([
       'uri' => 'public://example_1.jpg',
     ]);
     $image->save();
 
-    // Create a test style.
-    /** @var \Drupal\image\ImageStyleInterface $style */
-    $style = ImageStyle::create(['name' => 'main_style']);
-    $style->save();
-
     // Create an image item.
-    $alt = $this->randomString();
-    $title = $this->randomString();
-    $entity = EntityTest::create([
+    $this->entity = EntityTest::create([
```

Comment:
To clarify: This reuses the entity create code, but it does not reuse the entity - setUp() is run again for each test method.

Another way to use the entity create code is to have a createEntity() method, and then use a local variable where we need it. At least this way we only run it for those test methods that need it (currently all, but still)

Reply:
I agree as long as we can keep the random strings, we could pass them as parameters to the create 
```php
   $alt = $this->randomString();
    $title = $this->randomString();
    $entity = $this->createEntity($title, $alt);
```

Reply:
There's no need to have random strings, so moving the code to create the entity is a bit useless in this case. I'm a fan of the approach, but a `$this->createEntity()` in this case has no advantages.

Reply:
We discussed and we want to make sure that the class returns the correct alt and title so we will add the randomisation.

Reply:
Earlier I talked with @drishu and we identified 3 ways of structuring this to be able to reuse the entity creation code.
This is a recurring "problem" that we also see in other tests, so it is a good idea to agree on some guidelines.

1. One big test method to test both static factories. The entity can be created inside this method. Personally I don't like it as much.
2. Create the entity in `setUp()`, store in object property, then have two separate test methods both using `$this->entity`.
3. Create the entity in a separate method, have two separate test methods that use `$entity = $this->createEntity()`.

Personally I prefer option 3, because I generally like local variables and tend to avoid object properties.
This also guarantees that the creation only happens for those test methods where we need it.
(We don't save anything by having it in `setUp()`, because that is called again for each test method.)

I don't really care for the random values. In an earlier talk with @brummbar I thought he was the one who wants them :) For me, literal values are fine, perhaps better.

For literal values, I find it a bit weird if we set them explicitly inside a `testSomething()` method, but then we have the same hard-coded values in a separate assertion method.

Reply:
But, the above are only weak preferences, no strong opinions.

Reply:
I agreed that random doesn't bring much to this test so current implementation is ok